cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -188,7 +193,14 @@ private DatasetMeta getFromMds(DatasetId instance) throws Exception {
       // TODO: This shouldn't happen unless CDAP is in an invalid state - maybe give different error
       throw new NotFoundException(datasetTypeId);
     }
-    return new DatasetMeta(spec, typeMeta, null);
+    // for system dataset do not look up owner information in store as we know that it will be null.
+    // Also this is required to for cdap to start initially because we don't want to look up owner admin
```

Comment:
Did you mean:
```
This is required for CDAP to start, because initially we don't want to look up owner admin
(causing its lookup) as the SystemDatasetInitiator.getDataset is called when CDAPstarts
```
? (though I'm not sure even that is correct)